_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,626 @@
+additional signature adds about 18.25 vbytes, and there is at least 0.75
+vbytes of additional overhead from the opcode and extra parameters used
+to invoke multisig checking.
+
+    FIXME: plot similar to Newsletter #42
+
+## Complex spending with Taproot
+
+Bitcoin's Script language allows users to specify what conditions must
+be fulfilled in order for their bitcoins to be spent, conditions that
+sometimes require more than just signatures.  Taproot not only supports
+this feature but enhances it in several ways.
+
+We can consider this by looking at a basic Hash TimeLock Contract (HTLC)
+similar to those used in LN and cross-chain atomic swaps.  This contract
+can terminate three ways:
+
+1. Alice receives the contracted money (a payment) by publishing a
+   pre-image that releases the contract's hashlock.
+2. Bob receives the contracted money (a refund) after a timelock has
+   expired.
+3. Alice and Bob mutually agree about how to distribute the money,
+   usually because one of them could use one of the preceding
+   conditions to force a payment or refund.
+
+To take the LN case of HTLCs as an example, we can produce two
+independent scripts, each of which handles one of the first two items
+above:
+
+    (1) OP_HASH256 <hash> OP_EQUALVERIFY <Alice pubkey> OP_CHECKSIG
+    (2) <time delta> OP_CHECKSEQUENCEVERIFY OP_DROP <Bob pubkey> OP_CHECKSIG
+
+These separate scripts are then hashed so that they can be used as the
+leaves of a merkle tree.  As described earlier, the data to be hashed is
+prefixed with a tag (itself hashed) indicating its purpose.  "TapLeaf"
+is the tag in this case.  The leaves must also include the size of the
+script and a version (only version 0xc0 is defined in this proposal;
+other versions are available for future soft fork upgrades).
+
+After the two scripts are hashed, their digests are put in lexicographic
+order.  This ordering will allow later verification of a merkle
+inclusion proof without knowing whether or not each leaf or branch
+appeared in the tree to the left or right of its pair sibling, thus
+reducing the amount of data that needs to be communicated as well as
+potentially improving privacy.  After ordering, the two hashes will be
+hashed together with the prefix tag "TapBranch".  As this merkle tree
+only has two leaves, the resultant hash is the merkle root.
+
+    FIXME: diagram of tree
+
+This merkle tree only covers two of the HTLC's possible end states.  For
+the third case where Alice and Bob mutually agree on a spend, they can
+use signature aggregation using something like MuSig.  As in the
+multisig section above, they work together to create a single pubkey,
+called the *Taproot internal key.*
+
+The merkle root and the internal key are then hashed together (prefix
+tag, "TapTweak") and the resultant digest is used as a private key from
+which a pubkey is derived, this value being known as the *tweak.*  The
+tweak pubkey is added to the internal key (the same operation used in
```

Comment:
I don't understand the "(the same operation used in BIP32 HD key derivation)" and would suggest you drop it.